File: counting.java
import java.util.Arrays;

public class counting {
    static int[] countingSort(int[] array) {
        int max_value = Arrays.stream(array).max().orElse(Integer.MAX_VALUE);
        int[] count = new int[max_value + 1];
        int[] output = new int[array.length];

        for (int num : array) {
            count[num]++;
        }

        for (int i = 1; i <= max_value; i++) {
            count[i] += count[i - 1];
        }

        for (int i = array.length - 1; i >= 0; i--) {
            output[count[array[i]] - 1] = array[i];
            count[array[i]]--;
        }

        return output; // Retornar el array ordenado
    }

}

File: counting.py
def counting_sort(array):
    max_value = max(array)
    count = [0] * (max_value + 1)
    output = [0] * len(array)

    for num in array:
        count[num] += 1

    for i in range(1, len(count)):
        count[i] += count[i - 1]

    for num in reversed(array):
        output[count[num] - 1] = num
        count[num] -= 1

    return output
